### ISSUE TITLE
Any chance we could get a new release?

### ISSUE THREAD

**Issue by cshei** [mobile-shell/mosh#1115]
I use Mosh daily and really enjoy using it, but one downside of the current stable release is that it doesn't contain support for true colors, which has already been implemented and is available on the master branch.

Since I prefer using true colors in Vim, I usually end up patching Mosh with true color support on all my machines... it's trivial on Gentoo (with built-in support for user supplied patches in the package manager), but a bit more work on Fedora and others.

Since the last release of Mosh is over 3 years old, is there any chance we could get a stable release that has true color support?

Thanks in advance!

**Comment by cshei** [comment]
Friendly ping… judging from the 79 reactions on this, it seems that I'm not the only one interested in this 🙂

**Comment by GauthierPLM** [comment]
It seems more and more that Mosh is no longer maintained: issues are not answered, PR are not merged or reviewed, no commits on master for a long time,…

It does not make me super confident about the project future if no one takes over the code.

**Comment by keithw** [comment]
The truth is that our maintainer is on hiatus and we haven't made a release since 2017. However, our security track record remains infinitely better than any comparable project (OpenSSH, OpenSSL, etc.). We have never had a major security hole since Mosh 1.0 was released nine years ago. We're pretty proud of that! It's worth a considerable amount of confidence.

**Comment by GauthierPLM** [comment]
You can be proud of Mosh which is, indeed, a very good software, practical and much more flexible than SSH. It is all the more a pity that it does not evolve to be even better, even more reliable, and a software that continues to innovate to improve remote shells.

As the number of PRs and issues shows, people are willing to contribute and take the project further. I know how demanding it is to develop and maintain a software by yourself, so I understand the Mosh team might move away from producing code, but it's a pity that the team does not take advantage of these contributions if they participate in improving the software.

**Comment by AaronSwearingen** [comment]
Let's try a different angle to this: What is the release process and how is it initiated? What criteria must be met to move forward?

**Comment by karelbilek** [comment]
@GauthierPLM to play devil's advocate, maintaining secure and high-performant software is hard and you cannot just accept anything without carefully studying it.

so I kind of understand. Even if it breaks some stuff sometimes (see the issue above :( )

**Comment by fingolfin** [comment]
It's OK and normal for open source software maintainers to move on; there are many reasons (it has happened to myself several times).

But for such a popular project, it would be a shame to let it rot. Are there any attempts to find a new maintainer (or perhaps a team), who could be trusted by the current maintainers to step up and at least make a new release, merge some PRs etc.? 

I know it can be hard to let go, but the alternatives seem to be letting mosh die, or a fork, neither of which would be nice. Also, "letting go" doesn't have to be permanent, if the interest/motivation/time/whatever to work on mosh returns, they can do so, too, no?

**Comment by fiesh** [comment]
> @GauthierPLM to play devil's advocate, maintaining secure and high-performant software is hard and you cannot just accept anything without carefully studying it.

This is quite true.  However, this process should be done _prior_ to merging into `master`.  So once it's in `master`, making a release should be as simple as adding a tag.  And this is what this issue is about, the reason being that popular distributions will only follow releases.

**Comment by glensc** [comment]
I'm looking for a new release that builds with openssl 3.0.0

**Comment by vext01** [comment]
If mosh is rotting, has anyone tried [Eternal Terminal](https://github.com/MisterTea/EternalTerminal) as a replacement?

**Comment by t184256** [comment]
The problem being discussed isn't that mosh development isn't lively enough, but rather the contrary. We have 4 years worth of updates without so much as a patch version bump, and without version bumps maintainers don't update packages in distributions.

**Comment by vext01** [comment]
I'm an OpenBSD packager. Whilst we can backport patches if they are important enough, it's just a whole lot easier if there can be regular releases.

**Comment by msewell** [comment]
This is a duplicate of issue #974 (from April 2018), FYI – which only serves to cement the impression that no, there won't be a new release coming anytime soon (or ever).

**Comment by achernya** [comment]
We're gearing up to release a [mosh 1.4](https://mailman.mit.edu/pipermail/mosh-devel/2022-February/001598.html). As there hasn't been a release in a while, I anticipate it will take us a while to go through the existing commits and become confident it's good-to-go. Thank you for your patience!

**Comment by jonassmedegaard** [comment]
Thanks for the notice, @achernya - that it appreciated.  As is the actual work, obviously.

**Comment by laurentS** [comment]
Is there anything non security experts can do to help with a new release? Thanks for the great piece of software!

**Comment by RubenKelevra** [comment]
@laurentS sure, for example go through all the changes since the last release and check for functionality changes and cross-reference to the docs, if they are already updated :)

**Comment by vext01** [comment]
I see that `mosh-1.3.2.95rc1` has been tagged.

I gave it a superficial spin on OpenBSD and it appears to be working.

The only nit I have is that it now requires GNU make to build. Is that intentional?

```
$ make
...
Making all in include
Using $< in a non-suffix rule context is a GNUmake idiom (Makefile:543)
*** Error 1 in src (Makefile:390 'all-recursive': @fail=;  if (target_option=k; case ${target_option-} in  ?) ;;  *) echo "am__make_running_...)
```

**Comment by fiesh** [comment]
With mosh 1.4 having been tagged, I guess this can finally be closed!  (Although #1229 seems to hint towards colors still or again being broken -- if that's the case, most people who've been waiting for the new release will find this a bitter sweet moment ;))

**Comment by eminence** [comment]
Closing as complete 😄 🎉 

(Please open new issues for any new problems or questions)
